File: A_Panoramix_s_Prediction.py
n = int(input())
m = int(input())

next_prime = -1

for i in range(n + 1, 51):
    flag = True
    for j in range(2, i):
        if i % j == 0:
            flag = False
            break
    if flag:
        next_prime = i
        break

if next_prime == m:
    print("YES")
else:
    print("NO")

File: A_Panoramix_s_Prediction.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n, m;
    cin >> n >> m;

    int next = -1;

    for (int i = n + 1; i <= 50; i++)
    {
        bool flag = true;
        for (int j = 2; j < i; j++)
        {
            if (i % j == 0)
            {
                flag = false;
                break;
            }
        }
        if (flag)
        {
            next = i;
            break;
        }
    }

    if (next == m)
    {
        cout << "YES" << endl;
    }
    else
    {
        cout << "NO" << endl;
    }

    return 0;
}
